### DISCUSSION TITLE
Listen for change in selected device

### DISCUSSION THREAD

**Discussion by ClintH** [leolabs/ableton-js#92] [Q&A]
ableton-js v3.1.6

Hi!

This library is a treat, so well implemented! 👏

I was trying to find a way to be notified when the selected device is changed, regardless of track. Basically I want to try to control device parameters depending on what is selected.

There are some 'selected_' events on Ableton.song.view, but not 'selected_device' (or maybe what the LOM refers to as ['appointed_device'](https://docs.cycling74.com/max8/vignettes/live_object_model)?)

In the context of a Track instance, there is a 'devices' event, but this fires if a device is added or removed, nothing to do with selection.

I'm sure I've missed something obvious, but a search through the issues and discussions didn't yield anything.

thanks!

**Comment by leolabs** [comment]
Hey @ClintH,

Thanks for your feedback, I'm glad you like the library!

From what I can tell, the `selected_device` property lives in the `Track.View` class which AbletonJS hasn't implemented yet. I'll add it to my todo list and will implement support for it soon.

I just pushed an update that adds access to the `appointed_device` property, but I think that one is different from the selected device, or at least Live didn't emit any change events when I selected different devices.

Let me know if you have any further questions!

**Reply by ClintH** [reply]
Wow, what support! That's fantastic.  

I think there's real potential to use the library to overcome some of Live's MIDI mapping limitations. That's what I'm playing with at least...
